controllers/argocd/custommapper.go

```diff
@@ -266,3 +266,34 @@ func (r *ReconcileArgoCD) applicationSetSCMTLSConfigMapMapper(ctx context.Contex
 
 	return result
 }
+
+// namespaceResourceMapper maps a watch event on a namespaceManagement, back to the
+// ArgoCD object that we want to reconcile.
+func (r *ReconcileArgoCD) nmMapper(ctx context.Context, o client.Object) []reconcile.Request {
```

Comment:
At first I thought this logic was correct, but I think instead this function should reconcile EVERY `ArgoCD` CR on the cluster, rather than just the one that is pointed to by `.spec.ManagedBy`.

This allows us to catch the case where `.spec.managedBy` is changed from Argo CD namespace 'A', to Argo CD namespace 'B'.